en/docs/deploy/deploy-your-component.md

```diff
@@ -4,23 +4,90 @@ Deploying your component on Choreo makes it invocable. Once you have designed, t
+        Choreo requires you to perform the first deployment of each component manually so that you can provide values for any configurable variables that the implementation may include. Therefore, if you are deploying your component for the first time, you need to click **Deploy Manually** by default unless one or more of the following is true:<br/><br/>- The component is a REST API, REST API proxy, or GraphQL API, you can specify a sandbox endpoint to test the component in a sandbox environment if required. In such a scenario, you need to click **Configure & Deploy**.<br/><br/>- If you have included any [configurable variables]((https://ballerina.io/learn/by-example/configurable-variables/)) in the component implementation, you need to enter values for those variables. In such a scenario, you need to click **Configure & Deploy**.<br/><br/>- If you are deploying the component implementation from a Dockerfile, you would have selected **Docker** as the build preset when creating the component. In such a scenario, you need to click **Build and Deploy**. For more information, see [Deploy a Containerized Component](deploy-a-containerized-choreo-component.md).
+
+4. Enter values in the **Configure & Deploy** pane if required. This is required in the following scenarios:
+
+    - If your component implementation includes [configurable variables](#configure-variables), enter values for them.
+    - If your component is a REST API, REST API proxy, or GraphQL API, you can optionally provide a [sandbox endpoint](#enter-sandbox-endpoint) for testing purposes.
+    - If your component is a scheduled trigger, you need to enter the time interval at which the trigger should be activated.
 
-    ![Deploy component](../assets/img/tutorials/rest-api/deploy-api.png){.cInlineImage-full}
+6. If you do not want Choreo to automatically deploy the component after each commit that edits its implementation, toggle the **Auto Deploy on Commit** switch to disable automatic deployment,
 
-2. You can view the progress of the deployment from the console.
+7. Observe the progress of the deployment from the console.
 
     ![Deployment progress](../assets/img/tutorials/rest-api/deployment-progress.png){.cInlineImage-full}
 
-Once you deploy the component, the **Development** card indicates that it is active.
+Once you deploy the component, the **Development** card indicates that the component is active in the development environment.
 
 ![Deployed component](../assets/img/tutorials/rest-api/deployed-api.png){.cInlineImage-full}
 
 Now you can test your deployed component to check if it is working as expected.
+
+## Update an empty GitHub repository
+
+When a component with Ballerina as the build preset does not contain a file with the `.bal` in its GitHub path, the **Build Area** card displays the message shown in the image below.
+
+![Empty GitHub repository](../assets/img/deploy/empty-github-repository.png){.cInlineImage-full}
```

Comment:
we consider the repo empty if it has only *.md or .gitkeep or .gitignore or LICENCE files only. When there is a .bal file it doesn't show the repository as empty

Reply:
Corrected as specified.